### PR TITLE
feat(lb): add support for host_header & sni on healthchecks

### PR DESCRIPTION
**Pull request by yfodil**

### CODE REVIEW

File: scaleway/helpers_lb.go [hunk 1]
```diff
@@ -309,9 +310,10 @@ func expandLbHCHTTP(raw interface{}) *lbSDK.HealthCheckHTTPConfig {
 	}
 	rawMap := raw.([]interface{})[0].(map[string]interface{})
 	return &lbSDK.HealthCheckHTTPConfig{
-		URI:    rawMap["uri"].(string),
-		Method: rawMap["method"].(string),
-		Code:   expandInt32Ptr(rawMap["code"]),
+		URI:        rawMap["uri"].(string),
+		Method:     rawMap["method"].(string),
+		Code:       expandInt32Ptr(rawMap["code"]),
+		HostHeader: rawMap["host_header"].(string),
```
Comment (Monitob) [inline scaleway/helpers_lb.go:316]: Why this is keep it as `string`?  https://pkg.go.dev/github.com/scaleway/scaleway-sdk-go@v1.0.0-beta.13/api/lb/v1#HealthCheckHTTPConfig

File: scaleway/helpers_lb.go [hunk 2]
```diff
@@ -321,9 +323,11 @@ func flattenLbHCHTTPS(config *lbSDK.HealthCheckHTTPSConfig) interface{} {
...
 	return []map[string]interface{}{
 		{
-			"uri":    config.URI,
-			"method": config.Method,
-			"code":   flattenInt32Ptr(config.Code),
+			"uri":         config.URI,
+			"method":      config.Method,
+			"code":        flattenInt32Ptr(config.Code),
+			"host_header": config.HostHeader,
+			"sni":         config.Sni,
```
Comment (Monitob) [inline scaleway/helpers_lb.go:330]: Please verify this on the SDK. Sometimes is a [pointer](https://github.com/scaleway/scaleway-sdk-go/blob/5334351c3723/api/lb/v1/lb_sdk.go#L1658)

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/resource_lb_backend.go [hunk 3]
```diff
@@ -199,6 +199,11 @@ func resourceScalewayLbBackend() *schema.Resource {
 							Optional:    true,
 							Description: "The expected HTTP status code",
 						},
+						"host_header": {
```
[the remark this answers is not in the document]
Comment (yfodil) [reply scaleway/resource_lb_backend.go:202]: By definition URI identifies the specific resource being requested, the HTTP Host header identifies the server that is being requested and the domain name of the website being accessed.

File: scaleway/helpers_lb.go (hunk 1 above)
Comment (yfodil) [reply scaleway/helpers_lb.go:316]: I don't understand, it's a [string](https://pkg.go.dev/github.com/scaleway/scaleway-sdk-go@v1.0.0-beta.13/api/lb/v1#HealthCheckHTTPConfig) ?

File: scaleway/helpers_lb.go (hunk 2 above)
Comment (yfodil) [reply scaleway/helpers_lb.go:330]: The sni you are pointing is for the RouteMatch struct, where the one used im my case is for [HealthCheckHTTPSConfig](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/lb/v1/lb_sdk.go#L1288) struct, and its a string

**Review by Monitob — APPROVED** [review]

**Pull request merged**